Automattic/isolated-block-editor
# Collab: Prevent emoji from being separated by peer caret

Fixes #40

This fixes the issue in Chrome and Safari, but not in Firefox yet. An alternative approach is to use regex matching with a `u` (unicode-aware) flag, but it won't work with compound emoji like 👨‍👧‍👧. `Intl.Segmenter` is the only way to handle everything.

I don't think this bug is worth a polyfill, so let's go with the `Intl.Segmenter` approach and Firefox will get it when it [lands](https://bugzilla.mozilla.org/show_bug.cgi?id=1423593).

## Testing instructions

1. `yarn storybook`
2. Open two browser tabs with http://localhost:6006/?path=/story/collaboration--default
3. In one tab, type in something that ends with an emoji, like 🧛🏿‍♀️ or 👩‍👩‍👦 or 👍
4. The emoji should show up in the other tab correctly, without being broken apart by the peer caret.
